lms/djangoapps/courseware/grades.py

```diff
@@ -126,9 +127,18 @@ def grade(student, request, course, keep_raw_scores=False):
     """
     Wraps "_grade" with the manual_transaction context manager just in case
     there are unanticipated errors.
+    Send a signal to update the minimum grade requirement status.
     """
     with manual_transaction():
-        return _grade(student, request, course, keep_raw_scores)
+        grade_summary = _grade(student, request, course, keep_raw_scores)
+        MIN_GRADE_REQUIREMENT_STATUS.send_robust(
+            sender=None,
+            request=request,
```

Comment:
It looks like your receiver only needs the username, not the whole request.  Could we send that instead?